packages/dataviews/src/search-widget.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+import { matchSorter } from 'match-sorter';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useMemo, useDeferredValue } from '@wordpress/element';
+import {
+	BaseControl,
+	Icon,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { check, search } from '@wordpress/icons';
+
+export default function SearchWidget( { filter, view, onChangeView } ) {
+	const [ searchValue, setSearchValue ] = useState( '' );
+	const deferredSearchValue = useDeferredValue( searchValue );
+	const selectedFilter = view.filters.find(
+		( _filter ) => _filter.field === filter.field
+	);
+	const selectedValues = selectedFilter?.value;
+	const matches = useMemo( () => {
+		return matchSorter( filter.elements, deferredSearchValue ?? '', {
+			keys: [ 'label' ],
+		} );
+	}, [ filter.elements, deferredSearchValue ] );
+	return (
+		<Ariakit.ComboboxProvider
+			value={ searchValue }
+			setSelectedValue={ ( value ) => {
+				const currentFilter = view.filters.find(
+					( _filter ) => _filter.field === filter.field
+				);
+				const newFilters = currentFilter
+					? [
+							...view.filters.map( ( _filter ) => {
+								if ( _filter.field === filter.field ) {
+									return {
+										..._filter,
+										operator:
+											currentFilter.operator ||
+											filter.operators[ 0 ],
+										value,
+									};
+								}
+								return _filter;
+							} ),
+					  ]
+					: [
+							...view.filters,
+							{
+								field: filter.field,
+								operator: filter.operators[ 0 ],
+								value,
+							},
+					  ];
+				onChangeView( {
+					...view,
+					page: 1,
+					filters: newFilters,
+				} );
+			} }
+			setValue={ setSearchValue }
+		>
+			<BaseControl
+				className="dataviews-search-widget-filter-combobox__wrapper"
+				__nextHasNoMarginBottom
+			>
+				<div className="dataviews-search-widget-filter-combobox__input-wrapper">
+					<Ariakit.Combobox
```

Comment:
We should make sure the control is labelled, have looked into using `Ariakit.ComboboxLabel` ?